### ISSUE TITLE
Feature-Request: Allow passing requests verify option explicitly during instantiation of client

### ISSUE THREAD

**Issue by smallsam** [thycotic/python-tss-sdk#41]
It's common to require using a cacert bundle to connect to thycotic instances. Please allow passing verify option through to underlying requests calls (maybe use a requests session to persist it and other headers) so that it's easier to pass in.

I know it is possible to use the requests environment variable but this is trickier to manage when operating in a process with other ssl connections requiring their own specific ssl connection settings.
e.g.

`authorizer = PasswordGrantAuthorizer("https://hostname/SecretServer", "myusername", "mypassword", verify='/pathto_ca_bundle')`

**Comment by EndlessTrax** [comment]
Previously we have took the position that this is an environment issue, and it should be dealt with at that layer. However, you raise a good point regarding using multiple SSL connection settings.  So we have decided to reverse our decision and add this to the roadmap for development. We need to be sure we implement it carefully as this will also allow `verify=False` which is not something we want to encourage, for security reasons. 

Thank you for the request, @smallsam

**Comment by smallsam** [comment]
Absolutely, I think it's worthwhile passing verify through as is to allow for development use cases where verify=False might be useful.

I agree that you probably even want to actively discourage use of verify=False, perhaps by logging warnings when it is set, but on a quick inspection it doesn't look like you do any logging at all yet. Another one for the roadmap?

**Comment by EndlessTrax** [comment]
Yeah, we'll actively be discouraging the use of `verify=False` in the documentation, and we'll likely add a warning if used. 

Logging is also on the roadmap, too. Adding full CRUD is the priority at this point.
